code/modules/mob/living/carbon/alien/special/facehugger.dm

```diff
@@ -1,265 +1,267 @@
+		visible_message("<span class='danger'>[src] smashes against [target], but can't seem to latch on!</span>", "<span class='userdanger'>[src] smashes against you, but can't seem to latch on!</span>")
+		if(!sterile)
+			living_target.adjustBruteLoss(melee_damage_upper)
+			return TRUE
+	if(!iscarbon(target))
+		return
+	var/mob/living/carbon/carbon_target = target
+		
+	//Check for headlessness
+	if(!carbon_target.get_bodypart(BODY_ZONE_HEAD))
+		visible_message("<span class='danger'>[src] smashes against [target], but can't latch on!</span>", "<span class='userdanger'>[src] smashes against you, but cant latch on!</span>")
+		if(!sterile)
+			carbon_target.adjustBruteLoss(melee_damage_upper)
+		return FALSE
 
-//TODO: Make these simple_animals
-
-#define MIN_IMPREGNATION_TIME 100 //time it takes to impregnate someone
-#define MAX_IMPREGNATION_TIME 150
+	//Check for helmet
+	if(ishuman(carbon_target))
+		var/mob/living/carbon/human/human_target = carbon_target
+		if(human_target.is_mouth_covered(head_only = TRUE))
+			human_target.visible_message("<span class='danger'>[src] smashes against [target]'s [human_target.head]!</span>", "<span class='userdanger'>[src] smashes against your [human_target.head]!</span>")
+			COOLDOWN_START(src, coupling_cooldown, couple_retry_time)
+			return FALSE
 
-#define MIN_ACTIVE_TIME 200 //time between being dropped and going idle
-#define MAX_ACTIVE_TIME 400
+	//Check for mask
+	if(carbon_target.wear_mask)
+		var/obj/item/clothing/mask = carbon_target.wear_mask
+		if(istype(mask, /obj/item/clothing/mask/facehugger_item))
+			return FALSE
+		if(carbon_target.dropItemToGround(mask))
+			carbon_target.visible_message("<span class='danger'>[src] tears [mask] off of [target]'s face!</span>", "<span class='userdanger'>[src] tears [mask] off of your face!</span>")
+		else
+			carbon_target.visible_message("<span class='danger'>[src] tries to tear [mask] off of [target]'s face, but fails!</span>", "<span class='userdanger'>[src] trys to tear [mask] off of your face, but fails!</span>")
+			COOLDOWN_START(src, coupling_cooldown, couple_retry_time)
+			return FALSE
 
-/obj/item/clothing/mask/facehugger
+	target.visible_message("<span class='danger'>[src] leaps at [target]'s face!</span>", "<span class='userdanger'>[src] leaps at your face!</span>")
+	var/obj/item/clothing/mask/facehugger_item/hugger_item = BecomeItem()
+	if(!carbon_target.wear_mask && carbon_target.equip_to_slot_if_possible(hugger_item, ITEM_SLOT_MASK, FALSE, TRUE, TRUE))
+		hugger_item.Attach(carbon_target)
+	return TRUE
+
+/**
+ * Turns the facehugger into an item, storing the mob inside of it.
+ *
+ * Proc to turn the facehugger into an item and store the mob inside said item.
+ * The item version will take the name, description, and appearance of the facehugger.
+ */
+/mob/living/simple_animal/hostile/facehugger/proc/BecomeItem()
+	if(!mask_facehugger)
+		mask_facehugger = new(src)
+		mask_facehugger.facehugger_mob = src
+	mask_facehugger.forceMove(loc)
```

Comment:
Does the item have to be moved to `loc` ?

Reply:
In the instance where we already have a mask version, we need to get them out of ourselves because having something's location being inside something stored within in it, even for a moment, sounds like its going to break.